Document changes to instructions on setting up AWS and installing Stack (#48)
feature-creeps/observability-workshop: docs/create_vm_in_aws.md

# Creating a virtual machine using AWS for the purposes of installing the observability workshop stack

## Prerequisites

> > NOTE: These instructions are all written from the point of view of Amazon region `eu-west-2`. This is not required but please change each location of this region name if you choose to use a different one. If you encounter permissions issues, `sudo -i` should get you out of trouble (Anne-Marie, I can't find where in the instructions I ran into that!)

Reviewer: ```suggestion
> NOTE: These instructions are all written from the point of view of Amazon region `eu-west-2`. This is not required but please change each location of this region name if you choose to use a different one. If you encounter permissions issues, `sudo -i` will let you operate all the rest of the commands as a super user without having to set permissions on each request.
```

1. AWS account and developer command line credentials
    - Please follow along with AWS documentation if you have not yet created your own account.
    - Your default region is at the top right of the AWS console page next to your account name. 
    - Once you have an account set up, you will need to set up the [aws command line tool](https://docs.aws.amazon.com/cli/index.html).

>     - Somewhere along the line you will need to set environment variables with your AWS Access Key ID and Secret Key ID. You can find your Access Key ID on the [security_credentials](https://console.aws.amazon.com/iam/home#/security_credentials) page but you need to have saved your secret access key somewhere safe, or generate new ones.  

Reviewer: ```suggestion
    - Somewhere along the line you will need to set environment variables with your AWS Access Key ID and Secret Key ID. You can find your Access Key ID on the [security_credentials](https://console.aws.amazon.com/iam/home#/security_credentials) page but you need to have saved your secret access key somewhere safe, or [generate new credentials](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html).
```

2. A VPC (Virtual Private Cloud)
   - The AWS region you will be using needs an existing VPC in it for the application machine to be built in.
   - By default a VPC is created when you create a AWS account. This can be checked via command-line:
    ``` bash

>     `aws ec2 describe-vpcs`

Reviewer: ```suggestion
    aws ec2 describe-vpcs
```

    ```
    - If you wish to create a second VPC you can do so via command-line:
    ``` bash

>     `aws ec2 create-vpc --cidr-block 10.0.0.0/16`

Reviewer: ```suggestion
    aws ec2 create-vpc --cidr-block 10.0.0.0/16
```

    ```
3. An AMI (Amazon Machine Instance)
    - AMI's are region dependent. [Pick a Ubuntu flavour](https://cloud-images.ubuntu.com/locator/ec2/) Note: type in your region in the search, picking the first "bionic" one seems to work. Copy the AMI-ID, just the value of it, not the link address.

4. - use machine type `m5.2xlarge`
    - It is common for AWS to limit the larger machine types for newer accounts. You can check your machine type limits following [these](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-resource-limits.html) instructions.
    - If you do need to request a limit increase this can take a few days so be sure to request in the same region as your VPC to limit requesting a second time.
5. Honeycomb key

>    This stack will startup with or without a Honeycomb key. If you wish to integrate that data from the stack into Honeycomb you will need a key. Right now you need to ask one of the organisers for a key.

Reviewer: ```suggestion
   This stack will startup with or without a Honeycomb key. If you wish to integrate that data from the stack into Honeycomb you will need a key. Right now you need to ask one of the organisers for a key or signup for your own [free account](https://ui.honeycomb.io/signup/free) from which you can get your own key.
```
